src/solana/sdk/index.ts

```diff
@@ -1,51 +1,62 @@
-import { Connection, PublicKey } from '@solana/web3.js';
+import { web3 } from '@coral-xyz/anchor';
 import { BoringVaultSolana } from './boring-vault-solana';
 import { parseFullVaultData, FullVaultData } from './vault-state';
 import * as boringVaultIdl from './boring-vault-svm-idl.json';
 import { 
   AccountLayout
 } from '@solana/spl-token';
+import { createSolanaClient, type SolanaClient, Address } from 'gill';
 
 /**
  * Vault SDK adapter for mainnet testing
  * Wraps the existing BoringVaultSolana class
  */
 export class VaultSDK {
-  private connection: Connection;
+  private rpc: SolanaClient['rpc'];
   private boringVault: BoringVaultSolana;
-  private programId: PublicKey;
+  private programId: web3.PublicKey;
+  private solanaClient: SolanaClient;
   
-  constructor(connection: Connection) {
-    this.connection = connection;
+  constructor(urlOrMoniker: string) {
+    this.solanaClient = createSolanaClient({ urlOrMoniker });
+    this.rpc = this.solanaClient.rpc;
     
     // Get program ID from env or IDL
-    this.programId = new PublicKey(
+    this.programId = new web3.PublicKey(
       process.env.BORING_VAULT_PROGRAM_ID || 
       boringVaultIdl.address 
     );
     
-    // Initialize the BoringVaultSolana with the connection
+    // Initialize the BoringVaultSolana with the solanaClient
     this.boringVault = new BoringVaultSolana({
-      connection,
+      solanaClient: this.solanaClient,
       programId: this.programId.toString()
     });
   }
 
   /**
    * Get vault data for a given vault
    */
-  async getVaultData(vaultPubkey: PublicKey): Promise<FullVaultData> {
+  async getVaultData(vaultPubkey: web3.PublicKey): Promise<FullVaultData> {
+    // Convert web3.PublicKey to Address type for gill
+    const address = vaultPubkey.toBase58() as Address;
     // Get the account directly for the vaultPubkey parameter 
-    const accountInfo = await this.connection.getAccountInfo(vaultPubkey);
-    if (!accountInfo) {
+    const response = await this.rpc.getAccountInfo(
+      address,
+      { encoding: 'base64' }
+    ).send();
+    if (!response.value) {
```

Comment:
Should we also check for data length?


```suggestion
    if (!response.value || !response.value.data.length) {
```

Reply:
sure